Sources/Ink/Internal/Reader.swift

```diff
@@ -69,16 +69,25 @@ extension Reader {
         return count
     }
 
+    /// Read characters that match by evaluating a keypath
+    ///
+    /// - Parameters:
+    ///   - keyPath: A keypath to evaluate that is `true` for target characters.
+    ///   - maxCount: The maximum number of characters to attempt to read.
+    /// - Returns: The substring of characters successfully read
+    /// - Complexity: O(*n*), where *n* is the length of the string being read.
```

Comment:
Very nice that you're adding docs! I'll make a point to document the rest of `Reader`. I don't think we need to document all internal types, but since `Reader` is so central, I think it makes total sense 👍